gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/jsr223/GremlinScriptChecker.java

```diff
@@ -162,7 +168,10 @@ public class GremlinScriptChecker {
      * </pre>
      */
     private static final Pattern patternWithOptions =
-            Pattern.compile("\\.with\\(((?:" + timeoutTokens + "),(?<to>\\d*)(:?L|l)?)|((?:" + requestIdTokens + "),[\"'](?<rid>.*?))[\"']\\)");
+            Pattern.compile("\\.with\\((((?:"
```

Comment:
every time i touch this, i've double checked to make sure there is no significant performance penalty. this checker is executed a lot so we'd want to know if a slow down was happening here. could you just validate that there is no significant change with a quick benchmark? i don't expect anything given what i recall from the last update here, but i think it's worth continuing to validate assumptions on this one. maybe even make this a permanent benchmark in gremlin-benchmarks so that we dont have to start from scratch each time?

Reply:
please update the "explain" in the javadoc: https://regex101.com/

Reply:
"explain" updated in https://github.com/apache/tinkerpop/pull/1843/commits/b8582c750db1982a33ba41f4a9fda976ecc31eaa

Reply:
Benchmark                                                      Mode  Cnt       Score       Error  Units
GremlinScriptCheckerBenchmark.testParseAll                    thrpt   20  203909.105 ±  8915.473  ops/s
GremlinScriptCheckerBenchmark.testParseMaterializeProperties  thrpt   20  484647.497 ± 15267.578  ops/s
GremlinScriptCheckerBenchmark.testParseRequestId              thrpt   20  382702.687 ±  6140.784  ops/s

Reply:
benchmark added in https://github.com/apache/tinkerpop/pull/1843/commits/8c36b01b673982875d9d9e04d91f3ddce73860f5